src/Form/Serializer/BaseSerializerHandler.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Form\Serializer;
+
+use JMS\Serializer\Context;
+use JMS\Serializer\GraphNavigator;
+use JMS\Serializer\VisitorInterface;
+use Sonata\CoreBundle\Model\ManagerInterface;
+
+/**
+ * @author Sylvain Deloux <sylvain.deloux@ekino.com>
+ */
+abstract class BaseSerializerHandler implements SerializerHandlerInterface
+{
+    /**
+     * @var ManagerInterface
+     */
+    protected $manager;
+
+    /**
+     * @var string[]
+     */
+    protected static $formats;
+
+    public function __construct(ManagerInterface $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * @param string[] $formats
+     */
+    final public static function setFormats(array $formats)
+    {
+        static::$formats = $formats;
+    }
+
+    /**
+     * @param string $format
+     */
+    final public static function addFormat($format)
+    {
+        static::$formats[] = $format;
+    }
+
+    public static function getSubscribingMethods()
+    {
+        // NEXT_MAJOR : remove this block
+        if (null === static::$formats) {
+            static::$formats = ['json', 'xml', 'yml'];
+            @trigger_error(
+                '$formats has been set to default array("json", "xml", "yml"). Setting $formats to a
+                default array is deprecated since version 3.0 and will be removed in 4.0. Use SonataCoreBundle
+                configuration to add default serializer formats.',
+                E_USER_DEPRECATED
+            );
```

Comment:
```suggestion
            @trigger_error(sprintf(
                '%s::$formats has been set by default to array("json", "xml", "yml"). Ability to set them to a
                different value is deprecated since sonata-project/core-bundle 3.0 and will be removed in version 4.0. Use "sonata_core.serializer.formats" configuration instead.',
                __CLASS__
            ), E_USER_DEPRECATED);
```